### PR TITLE
Update workflow for updating Bicep types

### PR DESCRIPTION
**Pull request by ytimocin**
# Description

Updating the way we update bicep types.

## Type of change
- This pull request fixes a bug in Radius and has an approved issue (issue link required).

Fixes: #issue_number

### Summary
🆕🔄🎨

This pull request enhances the radius-bot functionality and improves the publish-bicep workflow. It adds a `/update-bicep` command to the `radius-bot.js` script that can update the extensibility provider types from a comment. It also simplifies and fixes the `.github/workflows/publish-bicep.yaml` file to use repository dispatch events and follow consistent style.

> _Oh, we are the radius-bot crew and we have a job to do_
> _We update the bicep types with a slash command or two_
> _We heave and ho and pull the rope, we make the workflow go_
> _We fix the quotes and indents too, we are the best, you know_

### Walkthrough
*  Add support for `/update-bicep` command to trigger extensibility provider types update from pull request comment ([link](https://github.com/radius-project/radius/pull/6584/files?diff=unified&w=0#diff-b94c0e124b5b65fb6fa8785928c5331547c124a96ecc225f93df9445a29a5109R50-R52), [link](https://github.com/radius-project/radius/pull/6584/files?diff=unified&w=0#diff-b94c0e124b5b65fb6fa8785928c5331547c124a96ecc225f93df9445a29a5109R135-R188))
*  Modify `publish-bicep.yaml` workflow to run on repository dispatch events of type `update-bicep` instead of pull request events ([link](https://github.com/radius-project/radius/pull/6584/files?diff=unified&w=0#diff-5dda38c017a17c459b16104779239cf891ceac425018e4b63263f4c7d773e061L19-R21))
*  Remove unnecessary token inputs for `actions/checkout@v2` and `peter-evans/create-pull-request@v3` actions in `publish-bicep.yaml` workflow ([link](https://github.com/radius-project/radius/pull/6584/files?diff=unified&w=0#diff-5dda38c017a17c459b16104779239cf891ceac425018e4b63263f4c7d773e061L80), [link](https://github.com/radius-project/radius/pull/6584/files?diff=unified&w=0#diff-5dda38c017a17c459b16104779239cf891ceac425018e4b63263f4c7d773e061L94))
*  Fix indentation and quote style in `publish-bicep.yaml` workflow ([link](https://github.com/radius-project/radius/pull/6584/files?diff=unified&w=0#diff-5dda38c017a17c459b16104779239cf891ceac425018e4b63263f4c7d773e061L7-R7), [link](https://github.com/radius-project/radius/pull/6584/files?diff=unified&w=0#diff-5dda38c017a17c459b16104779239cf891ceac425018e4b63263f4c7d773e061L29-R29), [link](https://github.com/radius-project/radius/pull/6584/files?diff=unified&w=0#diff-5dda38c017a17c459b16104779239cf891ceac425018e4b63263f4c7d773e061L40-R40))

### CODE REVIEW

File: .github/workflows/publish-bicep.yaml
Comment (vinayada1) [inline .github/workflows/publish-bicep.yaml:94]: I guess this will run in the context of the dispatch event. Might have to stay.

File: .github/scripts/radius-bot.js
Comment (vinayada1) [inline .github/scripts/radius-bot.js:177]: Fix comment

File: .github/scripts/radius-bot.js
Comment (kachawla) [inline .github/scripts/radius-bot.js:173]: nit: I don't think `testPayload` fits in this context. Maybe call it `requestPayload` or just `payload`?

```suggestion
        const requestPayload = {
            pull_head_ref: pull.data.head.sha,
            pull_head_repo: pull.data.head.repo.full_name,
            command: 'update-bicep',
            issue: issue,
        };
```

**Review by kachawla — COMMENTED** [review]
Will wait for more inputs on the offline discussion. My preference would be to make this workflow execution a requirement for all PRs (/PRs with generated type updates) by blocking the merge if it hasn't been run, instead of relying on approvers to remember it.

File: .github/scripts/radius-bot.js
Comment (shalabhms) [inline .github/scripts/radius-bot.js:178]: Does the owner permission gets in the context of this request ?

File: .github/workflows/publish-bicep.yaml
Comment (nithyatsu) [inline .github/workflows/publish-bicep.yaml:29]: was this changed needed?

File: .github/workflows/publish-bicep.yaml
Comment (nithyatsu) [inline .github/workflows/publish-bicep.yaml:33]: the edge branch seems to have v3, I am not sure I understand changing to v2, was it needed?

File: .github/workflows/publish-bicep.yaml
Comment (nithyatsu) [inline .github/workflows/publish-bicep.yaml:19]: nice 👍

File: .github/workflows/publish-bicep.yaml
Comment (AaronCrawfis) [reply .github/workflows/publish-bicep.yaml:33]: Let's use v3 as it addresses the node 12 deprecation

File: .github/workflows/publish-bicep.yaml
Comment (rynowak) [reply .github/workflows/publish-bicep.yaml:19]: Can we leave a big scary "thar be dragons here" comment about the security assumptions of `pull_request_target`?

Something like:

```
# Please be careful making changes to this workflow. This uses pull_request_target so it has access to our 
# secrets. This is OK because we're just copying files out of the contributors PR branch. 
#
# DO NOT run any code from the pull-request including makefile targets.
#
# See the official docs for more info:  https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_target
```

File: .github/workflows/publish-bicep.yaml
Comment (rynowak) [reply .github/workflows/publish-bicep.yaml:29]: We should be able to remove this, we don't use Go here.

**Review by shalabhms — DISMISSED** [review]
/lgtm,  however please resolve other 2 open review comments.

**Review by shalabhms — APPROVED** [review]

**Review by nithyatsu — APPROVED** [review]

**Review by rynowak — APPROVED** [review]

**Pull request merged**
